apps/root/src/pages/home/frame/index.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import { Grid, Typography } from 'ui-library';
+import Portfolio, { PortfolioRecord } from '../components/portfolio';
+import WalletSelector, { ALL_WALLETS } from '@common/components/wallet-selector';
+import { useAllBalances } from '@state/balances/hooks';
+import { formatUnits } from '@ethersproject/units';
+import CenteredLoadingIndicator from '@common/components/centered-loading-indicator';
+import Activity from '../components/activity';
+import { isUndefined } from 'lodash';
+
+const HomeFrame = () => {
+  const [selectedWalletOption, setSelectedWalletOption] = React.useState(ALL_WALLETS);
+  const { isLoadingAllBalances, ...allBalances } = useAllBalances();
+
+  const portfolioBalances = React.useMemo(
+    () =>
+      Object.values(allBalances).reduce<PortfolioRecord>((acc, { balancesAndPrices, isLoadingChainPrices }) => {
+        const newAcc = { ...acc };
+        Object.entries(balancesAndPrices).forEach(([tokenAddress, tokenInfo]) => {
+          const tokenKey = `${tokenInfo.token.chainId}-${tokenAddress}`;
+          newAcc[tokenKey] = {
+            price: tokenInfo.price,
+            token: tokenInfo.token,
+            balance: 0,
+            isLoadingPrice: isLoadingChainPrices,
+          };
+
+          Object.entries(tokenInfo.balances).forEach(([walletAddress, balance]) => {
+            const parsedBalance = parseFloat(formatUnits(balance, tokenInfo.token.decimals));
```

Comment:
lets use bignumber so we dont lose accuracy, you can use the `add` function from it